21.md

```diff
@@ -0,0 +1,185 @@
+LUD-21: Currencies in `payRequest`.
+=================================
+
+`author: lsunsi`
+`author: luizParreira`
+`author: lorenzolfm`
+
+---
+
+## Support for LNURL-pay currencies
+
+This document describes an extension to the [payRequest](https://github.com/lnurl/luds/blob/luds/06.md) base specification that allows the `WALLET` to send money to a `SERVICE` denominating the amount in a different currency. The features proposed enable many use cases ranging from denominating an invoice in a foreign currency to a remittance-like experience.
+
+The main features provided by this extension are:
+- `SERVICE` **MUST** inform `WALLET` what currencies it supports
+- `WALLET` **MAY** request an invoice with an amount denominated in one of the currencies
+- `WALLET` **MAY** request to the payment to be converted into one of the currencies
+
+The extension is opt-in and backward compatible. Further, a supporting `WALLET` can always tell if a `SERVICE` is also supporting beforehand so the communication is never ambiguous.
+
+### Wallet-side first request
+
+The first request is unchanged from the base specification.
+
+### Service-side first response
+
+`SERVICE` must alter its JSON response to the first request to include a `currencies` field, as follows:
+
+```typescript
+type BaseResponse = {
+    tag: "payRequest",
+    metadata: string,
+    callback: string,
+    maxSendable: number,
+    minSendable: number
+}
+
+type Currency = {
+  code: string, // Code of the currency, used as an ID for it. E.g.: BRL
+  name: string, // Name of the currency. E.g.: Reais
+  symbol: string, // Symbol of the currency. E.g.: R$
+  decimals: number, // Integer; Number of decimal places. E.g.: 2
```

Comment:
nit: It might be good to include a longer description about this field. With UMA, a bunch of folks integrating have been a bit confused by the mechanics around decimals/multiplier. Might be good to lay out some examples and show why decimals is important for clarity.

Reply:
FYI @lsunsi I added some examples and notes on small currency units in the uma spec that might be helpful here too - https://github.com/uma-universal-money-address/protocol/blob/main/umad-04-lnurlp-response.md#currency-examples

Reply:
This makes a lot of sense, I tried complementing the spec with this info and copied your example kind of . What do you think of (https://github.com/lnurl/luds/pull/251/commits/6a5b6e029f7716a6a6caec75b8b2521fef29d148)?

Reply:
Cool, that captures the gist of it! I think the main difference is that UMA has an explicit max of 8 decimals, but keeping it open is fine too. LGTM